src/main/java/redis/clients/jedis/csc/ClientSideCache.java

```diff
@@ -1,120 +1,210 @@
 package redis.clients.jedis.csc;
 
 import java.nio.ByteBuffer;
+import java.util.ArrayList;
+import java.util.Collection;
 import java.util.List;
 import java.util.Map;
-import java.util.Objects;
 import java.util.Set;
 import java.util.concurrent.ConcurrentHashMap;
-import java.util.function.Function;
+import java.util.concurrent.locks.ReentrantLock;
+import java.util.stream.Collectors;
 
-import redis.clients.jedis.CommandObject;
 import redis.clients.jedis.annots.Experimental;
 import redis.clients.jedis.util.SafeEncoder;
 
 /**
- * The class to manage the client-side caching. User can provide any of implementation of this class to the client
- * object; e.g. {@link redis.clients.jedis.csc.CaffeineClientSideCache CaffeineClientSideCache} or
- * {@link redis.clients.jedis.csc.GuavaClientSideCache GuavaClientSideCache} or a custom implementation of their own.
+ * The class to manage the client-side caching. User can provide any of implementation of this class
+ * to the client object; e.g. {@link redis.clients.jedis.csc.CaffeineClientSideCache
+ * CaffeineClientSideCache} or {@link redis.clients.jedis.csc.GuavaClientSideCache
+ * GuavaClientSideCache} or a custom implementation of their own.
  */
 @Experimental
-public abstract class ClientSideCache {
+public abstract class ClientSideCache implements Cache {
```

Comment:
IMO, we shouldn't use both ClientSideCache and Cache here. Use `AbstractCache` or similar.

Reply:
sure,, AbstractCache sounds good.